src/test/librbd/test_mock_Journal.cc

```diff
@@ -186,6 +186,10 @@ OpenRequest<MockJournalImageCtx> *OpenRequest<MockJournalImageCtx>::s_instance =
 } // namespace librbd
 
 // template definitions
+
+#include "librbd/journal/PromoteRequest.cc"
+template class librbd::journal::PromoteRequest<librbd::MockJournalImageCtx>;
```

Comment:
... actually, never mind -- you should just provide a dummy (specialized) version above like the other classes since we only want to include the code definitions for the unit under test